### PR TITLE
Fix wait functionality in svcat register command

### PR DESCRIPTION
**Pull request by jberkhahn**
Never actually made this work before, this should fix that.

### CODE REVIEW

**Comment by jberkhahn** [comment]
I think this is barfing on a data race that I've included (intentionally) to test the wait functionality. Is there some way to fix that?

File: pkg/svcat/service-catalog/broker.go [hunk 1]
```diff
@@ -88,7 +92,7 @@ func (sdk *SDK) RetrieveBrokers(opts ScopeOptions) ([]Broker, error) {
 func (sdk *SDK) RetrieveBroker(name string) (*v1beta1.ClusterServiceBroker, error) {
 	broker, err := sdk.ServiceCatalog().ClusterServiceBrokers().Get(name, v1.GetOptions{})
 	if err != nil {
-		return nil, fmt.Errorf("unable to get broker '%s' (%s)", name, err)
+		return nil, errors.Wrapf(err, "unable to get broker '%s' (%s)", name, err)
```
Comment (jboyd01) [inline pkg/svcat/service-catalog/broker.go]: when using errors.Wrapf, the error message is concatenated with your format text, so you don't need the final (%s), and err param - - the text from the original error is automatically included and using what you have in here now will duplicate the error text.  I used this and Carolyn pointed out this issue.

File: pkg/svcat/service-catalog/service-catalogfakes/fake_svcat_client.go [hunk 2]
```diff
@@ -2744,6 +2814,8 @@ func (fake *FakeSvcatClient) Invocations() map[string][][]interface{} {
 	defer fake.retrieveClassByIDMutex.RUnlock()
 	fake.retrieveClassByPlanMutex.RLock()
 	defer fake.retrieveClassByPlanMutex.RUnlock()
+	fake.createClassMutex.RLock()
+	defer fake.createClassMutex.RUnlock()
```
Comment (jboyd01) [inline pkg/svcat/service-catalog/service-catalogfakes/fake_svcat_client.go]: Are these two createClassMutex's supposed to be included here?

Comment (jberkhahn) [reply pkg/svcat/service-catalog/service-catalogfakes/fake_svcat_client.go]: This isn't from my change - it looks like someone mucked with the interface and didn't regenerate the fake.

File: pkg/svcat/service-catalog/broker.go (hunk 1 above)
Comment (jberkhahn) [reply pkg/svcat/service-catalog/broker.go]: Fixed this

File: pkg/svcat/service-catalog/broker.go [hunk 3]
```diff
@@ -88,7 +92,7 @@ func (sdk *SDK) RetrieveBrokers(opts ScopeOptions) ([]Broker, error) {
 func (sdk *SDK) RetrieveBroker(name string) (*v1beta1.ClusterServiceBroker, error) {
 	broker, err := sdk.ServiceCatalog().ClusterServiceBrokers().Get(name, v1.GetOptions{})
 	if err != nil {
-		return nil, fmt.Errorf("unable to get broker '%s' (%s)", name, err)
+		return nil, errors.Wrapf(err, "unable to get broker '%s'", name)
```
Comment (carolynvs) [inline pkg/svcat/service-catalog/broker.go]: 💯

**Pull request merged**